src/community/csw-iso/src/main/java/org/geoserver/csw/records/iso/FeatureCatalogueDescriptor.java

```diff
@@ -0,0 +1,143 @@
+package org.geoserver.csw.records.iso;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Collections;
+import java.util.List;
+import net.opengis.cat.csw20.ElementSetType;
+import org.geoserver.csw.records.AbstractRecordDescriptor;
+import org.geoserver.csw.records.RecordFeatureTypeRegistryConfiguration;
+import org.geoserver.platform.GeoServerExtensions;
+import org.geotools.data.Query;
+import org.geotools.data.complex.feature.type.FeatureTypeRegistry;
+import org.geotools.data.complex.util.EmfComplexFeatureReader;
+import org.geotools.feature.NameImpl;
+import org.geotools.feature.type.FeatureTypeFactoryImpl;
+import org.geotools.xsd.SchemaIndex;
+import org.opengis.feature.type.AttributeDescriptor;
+import org.opengis.feature.type.FeatureType;
+import org.opengis.feature.type.FeatureTypeFactory;
+import org.opengis.feature.type.Name;
+import org.opengis.filter.Filter;
+import org.opengis.filter.expression.PropertyName;
+import org.xml.sax.helpers.NamespaceSupport;
+
+/**
+ * Describes the ISO FeatureCatalogue records and provides some handy constants to help building
+ * features representing FC_FeatureCatalogue.
+ *
+ * @author Niels Charlier
+ */
+public class FeatureCatalogueDescriptor extends AbstractRecordDescriptor {
+
+    public static FeatureType FEATURECATALOGUE_TYPE;
+    public static AttributeDescriptor FEATURECATALOGUE_DESCRIPTOR;
+
+    static {
+        FeatureTypeFactory typeFactory = new FeatureTypeFactoryImpl();
+
+        EmfComplexFeatureReader reader = EmfComplexFeatureReader.newInstance();
+
+        SchemaIndex index = null;
```

Comment:
I would add a log here, usually querying external URLs are a source of issues.

Reply:
as said above, it reads it actually finds it locally using resolver